client.go

```diff
@@ -1,20 +1,32 @@
-package redis
+package dpredis
 
 import (
-	"context"
+	"encoding/json"
 	"errors"
-	"github.com/ONSdigital/log.go/log"
+	"fmt"
 	"time"
 
+	. "github.com/ONSdigital/dp-redis/interfaces"
 	"github.com/ONSdigital/dp-sessions-api/session"
-	goredis "github.com/go-redis/redis"
+	"github.com/go-redis/redis"
 )
 
-var ctx = context.Background()
+var (
+	ErrEmptySessionID = errors.New("session id required but was empty")
+	ErrEmptySession   = errors.New("session is empty")
+	ErrEmptyAddress   = errors.New("address is empty")
+	ErrEmptyPassword  = errors.New("password is empty")
+	ErrInvalidTTL     = errors.New("ttl should not be zero")
+)
+
+// RedisClient - structure for the redis client
+type RedisClient struct {
+	client RedisClienter
+}
 
-// Client - structure for the redis client
+// Client - structure for the cache client
 type Client struct {
-	client *goredis.Client
+	client RedisClient
```

Comment:
Is having a type for this really necessary? i'm not sure what value it is adding, when the interface itself could be used for this type. You could implement the necessary functions off the Client type, without the extra hop through RedisClient

Reply:
Yup, agree. Looking at it again it's a bit convoluted having this too. I think initially it had some other stuff in there too but it had been reworked.